nixos/modules/services/monitoring/do-agent.nix

```diff
@@ -0,0 +1,53 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.do-agent;
+in
+{
+  options.services.do-agent = {
+    enable = mkEnableOption "do-agent, the DigitalOcean droplet metrics agent";
+
+    user = mkOption {
+      type = types.str;
+      default = "do-agent";
+      description = "User account under which do-agent runs.";
+    };
+
+    group = mkOption {
```

Comment:
Is there a need for user/group to be configurable? Also try to see if `DynamicUser` works for this.